library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -54,54 +65,51 @@ class FirebaseAnalyticsService @VisibleForTesting internal constructor(
-        if (event.isForSystem(AnalyticsSystem.FIREBASE)) when (event) {
+        if (event.isForSystem(AnalyticsSystem.FIREBASE) || event.isForSystem(AnalyticsSystem.ADOBE)) when (event) {
             is AnalyticsScreenEvent -> handleScreenEvent(event)
             is AnalyticsActionEvent -> handleActionEvent(event)
         }
     }
 
-    // region ActivityLifecycleCallbacks
-    init {
-        app.registerActivityLifecycleCallbacks(this)
-    }
-
-    private var currentActivity: Activity? by weak()
-
-    override fun onActivityResumed(activity: Activity) {
-        currentActivity = activity
-    }
-
-    override fun onActivityPaused(activity: Activity) {
-        currentActivity = currentActivity?.takeUnless { it === activity }
-    }
-
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
-    override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
-    override fun onActivityDestroyed(activity: Activity) = Unit
-    // endregion ActivityLifecycleCallbacks
-    // endregion Tracking Events
-
     @MainThread
     private fun handleScreenEvent(event: AnalyticsScreenEvent) {
-        currentActivity?.let { firebase.setCurrentScreen(it, event.screen, null) }
+        val bundle = Bundle().apply {
+            putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
+            putString(USER_PROP_APP_NAME, VALUE_GODTOOLS)
+            event.locale?.let { putString(USER_PROP_CONTENT_LANGUAGE, LocaleCompat.toLanguageTag(it)) }
+            event.adobeSiteSection?.let { putString(SITE_SECTION, event.adobeSiteSection) }
+            event.adobeSiteSubSection?.let { putString(SITE_SUB_SECTION, event.adobeSiteSubSection) }
+        }
+        firebase.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW, bundle)
     }
 
     @MainThread
     private fun handleActionEvent(event: AnalyticsActionEvent) {
-        firebase.logEvent(event.firebaseEventName, null)
+        val bundle = Bundle().apply {
+            event.adobeAttributes?.forEach { attribute ->
+                val attributeKey = attribute.key.replace(Regex("[ \\-.]"), "_")
+                putString(attributeKey, attribute.value.toString())
+            }
+        }
+
+        firebase.logEvent(event.firebaseEventName, bundle)
     }
 
     init {
-        oktaUserProfileProvider.userInfoFlow()
-            .map { it?.ssoGuid }
-            .distinctUntilChanged()
-            .onEach { firebase.setUserId(it) }
+        oktaUserProfileProvider.userInfoFlow(refreshIfStale = false)
+            .onEach { it ->
+                firebase.setUserProperty(USER_PROP_LOGGED_IN_STATUS, "${it != null}")
+                firebase.setUserProperty(USER_PROP_APP_NAME, VALUE_GODTOOLS)
+                firebase.setUserProperty(USER_PROP_CONTENT_LANGUAGE, LocaleCompat.toLanguageTag(Locale.getDefault()))
```

Comment:
is this meant to be the device language or the user's selected primary language?

Reply:
I believe the device, that's what we did for adobe.

Reply:
actually we never had the device language sent for adobe, we were only sending the content language from within individual events. (that's already in place [here](https://github.com/CruGlobal/godtools-android/pull/1396/files#diff-16cc0dd8554c9cc5ad5d84d969fb2c391765505cce3113ebc3537609ffe13763R79)).

I won't block on this because it is different data, it just happened to be new data vs what we had before. I know this would be useful for MPDX/MH, but not nearly as useful in GodTools since most of the app is actually driven by the primary language and not the device language

Reply:
Oh, you're right, this is how it's done for MPDx. Yeah, we'll see if they end up using it and if not we can remove it in the clean-up. But I agree I don't see the need either.